backend/server/schema.py

```diff
@@ -28,20 +36,130 @@ class Meta:
+        description=(
+            "Stats for predictions made by the given model through the given round"
+        ),
+    )
+
+    def resolve_round_number(self, _info):
+        return self.get("match__round_number")
+
+    def resolve_model_predictions(self, _info):
+        return [
+            {"model_name": model_name, "cumulative_correct_count": correct_count}
+            for model_name, correct_count in self.items()
+            if model_name != "match__round_number"
+        ]
+
+
+class YearlyPredictionsType(graphene.ObjectType):
+    """Model prediction stats per year"""
+
+    prediction_model_names = graphene.List(
+        graphene.String, description="All model names available for the given year"
+    )
+    predictions_by_round = graphene.List(
+        RoundPredictionType, description=("Predictions for the year grouped by round")
+    )
+
+    def resolve_prediction_model_names(self, _info):
+        return self.distinct("ml_model__name").values_list("ml_model__name", flat=True)
+
+    def resolve_predictions_by_round(self, _info):
+        query_set = (
+            self.values("match__round_number", "ml_model__name")
+            .order_by("match__round_number")
+            .annotate(correct_count=Count("is_correct", filter=Q(is_correct=True)))
+        )
+
+        calculate_cumulative_correct = (
+            lambda df: df.groupby("ml_model__name")
+            .expanding()
+            .sum()
+            .reset_index(level=0)["correct_count"]
+            .rename("cumulative_correct_count")
+        )
+
+        # TODO: There's definitely a way to do these calculations via SQL, but chained
+        # GROUP BYs and calculations based on calculations is a bit much for me
+        # right now, so I'll come back and figure it out later
+        return (
+            pd.DataFrame(list(query_set))
+            .assign(cumulative_correct=calculate_cumulative_correct)
+            .pivot(
+                columns="ml_model__name",
+                values="cumulative_correct",
+                index="match__round_number",
+            )
+            .reset_index()
+            .fillna(0)
+            .to_dict("records")
+        )
 
-class Query(graphene.ObjectType):
-    predictions = graphene.List(PredictionType, year=graphene.Int(default_value=None))
 
-    def resolve_predictions(self, _info, year=None):  # pylint: disable=R0201
+class Query(graphene.ObjectType):
+    predictions = graphene.List(
+        PredictionType,
+        year=graphene.Int(default_value=None),
+        description="Basic prediction type based on Prediction data model",
+    )
+
+    prediction_years = graphene.List(
+        graphene.Int,
+        description="All years for which model predictions exist in the database",
+    )
+
+    yearly_predictions = graphene.Field(
+        YearlyPredictionsType,
+        year=graphene.Int(),
+        description="Model prediction stats per year",
+    )
+
+    def resolve_predictions(self, _info, year=None):
```

Comment:
is this the old graphql predictions endpoint? do you want to keep it?

Reply:
Same as above